core/src/main/java/nl/nn/adapterframework/http/HttpSenderBase.java

```diff
@@ -259,18 +271,20 @@ protected URI getURI(String url) throws URISyntaxException, UnsupportedEncodingE
 			uri.setPath("/");
 		}
 
-		// Encode query param values.
-		ArrayList<NameValuePair> pairs = new ArrayList<>(uri.getQueryParams().size());
-		for(NameValuePair pair : uri.getQueryParams()) {
-			String paramValue = pair.getValue(); //May be NULL
-			if(StringUtils.isNotEmpty(paramValue)) {
-				paramValue = URLEncoder.encode(paramValue, getCharSet()); //Only encode if the value is not null
+		if(encodeQueryParameters) {
+			// Encode query param values.
+			ArrayList<NameValuePair> pairs = new ArrayList<>(uri.getQueryParams().size());
+			for(NameValuePair pair : uri.getQueryParams()) {
+				String paramValue = pair.getValue(); //May be NULL
+				if(StringUtils.isNotEmpty(paramValue)) {
+					paramValue = URLEncoder.encode(paramValue, getCharSet()); //Only encode if the value is not null
+				}
+				pairs.add(new BasicNameValuePair(pair.getName(), paramValue));
+			}
+			if(pairs.size() > 0) {
+				uri.clearParameters();
+				uri.addParameters(pairs);
```

Comment:
Ik denk dat dit de verkeerde plek is om parameters te URLEncoden. Volgens mij moet je het hier niet doen, en wel in HttpSender.getMethod(). Dan hoef je er ook geen switch voor te maken.